lib/samples/show_service_area/show_service_area_sample.dart

```diff
@@ -80,84 +136,124 @@ class _ShowServiceAreaSampleState extends State<ShowServiceAreaSample> {
-      List<ServiceAreaFacility> facilities) {
-    _serviceAreaParameters.setFacilities(facilities);
+      // for each graphic in the facilities graphics overlay, add a facility to the parameters.
+      final facilities = _facilityGraphicsOverlay.graphics
+          .map((graphic) =>
+              ServiceAreaFacility(point: graphic.geometry as ArcGISPoint))
+          .toList();
+      _serviceAreaParameters.setFacilities(facilities);
 
-    return _serviceAreaTask.solveServiceArea(
-      serviceAreaParameters: _serviceAreaParameters,
-    );
-  }
+      // for each graphic in the barriers graphics overlay, add a polygon barrier to the parameters.
+      final barriers = _barrierGraphicsOverlay.graphics
+          .map(
+              (graphic) => PolygonBarrier(polygon: graphic.geometry as Polygon))
+          .toList();
+      _serviceAreaParameters.setPolygonBarriers(barriers);
 
-  void displayServiceArea(ServiceAreaResult result) {
-    // Get the polygons for the facility. In this sample, there will only be one.
-    final serviceAreaPolygons = result.getResultPolygons(facilityIndex: 0);
-
-    final serviceAreaGraphics = serviceAreaPolygons.map((serviceAreaPolygon) {
-      return Graphic(
-        geometry: serviceAreaPolygon.geometry,
-        attributes: {
-          'fromImpedanceCutoff': serviceAreaPolygon.fromImpedanceCutoff,
-          'toImpedanceCutoff': serviceAreaPolygon.toImpedanceCutoff,
-        },
-        symbol: _serviceAreaSymbol,
+      // solve the service area using the parameters.
+      final serviceAreaResult = await _serviceAreaTask.solveServiceArea(
+        serviceAreaParameters: _serviceAreaParameters,
       );
-    });
 
-    _serviceAreaGraphicsOverlay.graphics.addAll(serviceAreaGraphics);
+      // display service area polygons for each facility - since the service area parameters have
+      // geometryAtOverlap set to dissolve and the impedence cutoff values are the same across facilities,
+      // we only need to draw the joined polygons for one of the facilities.
+      final serviceAreaPolygons =
+          serviceAreaResult.getResultPolygons(facilityIndex: 0);
+      for (var i = 0; i < serviceAreaPolygons.length; i++) {
+        _serviceAreaGraphicsOverlay.graphics.add(
+          Graphic(
+            geometry: serviceAreaPolygons[i].geometry,
+            symbol: _serviceAreaSymbols[i],
+          ),
+        );
+      }
+
+      // re-enable the UI once the service area task is finished.
+      setState(() {
+        _ready = true;
+      });
+    } else {
+      if (mounted) {
+        showDialog(
+          context: context,
+          builder: (context) => AlertDialog(
+            content: const Text(
+                'At least 1 facility is required to perform a service area calculation.'),
+            actions: [
+              TextButton(
+                  onPressed: () => Navigator.pop(context, 'OK'),
+                  child: const Text('OK'))
```

Comment:
Add trailing comma after Text('OK')

Reply:
I think we've decided it's fine not to if it's on one/two lines with regards to trailing commas. @PaulAllanSturm can you confirm that goes here?

Reply:
The `TextButton(...)` declaration doesn't fit on two lines, so its parameter list should have a trailing comma.